programs/uxd/tests/integration_tests/suites/test_identity_depository_mint_and_redeem.rs

```diff
@@ -0,0 +1,245 @@
+use solana_program_test::tokio;
+use solana_sdk::signer::keypair::Keypair;
+use solana_sdk::signer::Signer;
+
+use crate::integration_tests::api::program_spl;
+use crate::integration_tests::api::program_test_context;
+use crate::integration_tests::api::program_uxd;
+
+#[tokio::test]
+async fn test_identity_depository_mint_and_redeem(
+) -> Result<(), program_test_context::ProgramTestError> {
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Setup basic context and accounts needed for this test suite
+    // ---------------------------------------------------------------------
+
+    let mut program_test_context = program_test_context::create_program_test_context().await;
+
+    // Fund payer
+    let payer = Keypair::new();
+    program_spl::instructions::process_lamports_airdrop(
+        &mut program_test_context,
+        &payer.pubkey(),
+        1_000_000_000_000,
+    )
+    .await?;
+
+    // Create the program keys structure (find/create all important keys)
+    let program_keys = program_uxd::accounts::create_program_keys();
+
+    // Initialize basic UXD program state
+    program_uxd::procedures::process_deploy_program(
+        &mut program_test_context,
+        &program_keys,
+        &payer,
+    )
+    .await?;
+
+    // Main actor
+    let user = Keypair::new();
+
+    // Create a collateral account for our user
+    let user_collateral = program_spl::instructions::process_associated_token_account_get_or_init(
+        &mut program_test_context,
+        &payer,
+        &program_keys.collateral_mint.pubkey(),
+        &user.pubkey(),
+    )
+    .await?;
+    // Create a redeemable account for our user
+    let user_redeemable = program_spl::instructions::process_associated_token_account_get_or_init(
+        &mut program_test_context,
+        &payer,
+        &program_keys.redeemable_mint,
+        &user.pubkey(),
+    )
+    .await?;
+
+    // Useful amounts used during testing scenario
+    let amount_we_use_as_supply_cap = 50_000_000;
```

Comment:
We could improve using a conversion function "scale(50, USDC_DECIMALS)" which would be more readable for humans (gives you the amount and currency at a glance without mind gymnastics)

Reply:
Sounds good, ill do that in next incremental PR